specification/security.md

```diff
@@ -0,0 +1,243 @@
+
+<figure style="text-align:center">
+  <object type="image/svg+xml" data="{{site.baseurl}}/img/security-object.svg"></object>
+  <figcaption>Security Scheme objects are referenced as Security Requirements, either <a href="structure.html#minimal-openapi-description-structure">globally</a> or by an <a href="paths.html#the-operation-object">Operation</a>.</figcaption>
+</figure>
+
+A Security Requirement declared for a given Operation takes precedence over global Security Requirements. A number of security mechanisms are supported. Each mechanism is indicated using the `type` property, which is shown in the examples below.
+
+There are currently five supported security types, namely:
+
+- API Keys
+- HTTP Authentication
+- Mutual TLS
+- OAuth 2.0
+- OpenID Connect
+
+Each is discussed in the sections below.
+
+## API Keys
+
+An API key is - generally speaking - the API equivalent of a password and is used to supply a secret known only by a given API consumer and the API provider. There are no formal standards that govern API keys and the industry generally relies on accepted best practices for their definition. OpenAPI therefore provides flexibility in how they are defined.
+
+For example, an API provider might use the HTTP header `api-key` as the means for API consumers to send their API keys with each request. A Security Scheme object can be defined to indicate in which HTTP header the API Key should be sent:
+
+```yaml
+components:
+  securitySchemes:
+    defaultApiKey:
+      description: API key provided in console
+      type: apiKey
+      name: api-key
+      in: header
+```
+
+The API provider can then use the name of this Scheme Scheme object to apply security either globally or for a given Operation (both are shown below for the sake of exemplar):
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Tic Tac Toe
+  description: |
+    This API allows writing down marks on a Tic Tac Toe board
+    and requesting the state of the board or of individual squares.
+  version: 1.0.0
+security:
+  defaultApiKey: []
+paths:
+  /board:
+    get:
+      security:
+        defaultApiKey: []
+```
+
+This method of referencing Security Scheme objects is valid for all types. The array shown as the value of `defaultApiKey` above is populated for OAuth Flow and OpenID Connect objects, which have some additional features that are discussed below. An empty array is provided in all other cases.
+
+> Please note that correctly resolving Security Requirements may be impacted when merging multiple OpenAPI documents. You should check your tooling provider's method before assuming a specific resolution approach.
```

Comment:
@handrews added this note as per previous comments on Security Requirement resolution

Reply:
Thanks!  I try to avoid the term "merging" because it makes people think of things like [JSON Merge Patch](https://www.rfc-editor.org/rfc/rfc7396.html) and the old `$merge` proposal which is not how OAS works (although Overlays might).  I'd just go with "combining" or "referencing".

Reply:
Since we've defined the terms "entry document" and "referenced document" in #79, you could just say:

> Please note that different tools may resolve Security Requirement Objects in referenced documents differently.  You shouldl check...

Reply:
👍 gone with your suggestion as I think it sounds nicerer

Reply:
@SensibleWood great, thanks!  All looks great to me now.